Extension/src/Debugger/configurationProvider.ts

```diff
@@ -257,16 +257,9 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
         let buildTasks: CppBuildTask[] = [];
         await this.loadDetectedTasks();
         // Remove the tasks that are already configured once in tasks.json.
-        const dedupDetectedBuildTasks: CppBuildTask[] = DebugConfigurationProvider.detectedBuildTasks.filter(taskDetected => {
-            let isAlreadyConfigured: boolean = false;
-            for (const taskJson of configuredBuildTasks) {
-                if ((taskDetected.definition.label as string) === (taskJson.definition.label as string)) {
-                    isAlreadyConfigured = true;
-                    break;
-                }
-            }
-            return !isAlreadyConfigured;
-        });
+        const dedupDetectedBuildTasks: CppBuildTask[] = DebugConfigurationProvider.detectedBuildTasks.filter(taskDetected =>
+            (configuredBuildTasks ? !configuredBuildTasks.some(taskJson => (taskJson.definition.label === taskDetected.definition.label)) : true));
```

Comment:
you should be able to simplify this to not need the `? : true` by just returning `!configuredBuildTasks?.some(...)` instead. (`?.` replaces the `.`)